src/calliope/model.py

```diff
@@ -93,14 +89,22 @@ def __init__(
         if isinstance(model_definition, xr.Dataset):
             self._init_from_model_data(model_definition)
         else:
-            (model_def, self._model_def_path, applied_overrides) = (
-                load.load_model_definition(
-                    model_definition, scenario, override_dict, **kwargs
-                )
+            if isinstance(model_definition, dict):
+                model_def_dict = AttrDict(model_definition)
+            else:
+                self._def_path = str(model_definition)
+                model_def_dict = AttrDict.from_yaml(model_definition)
+
+            (model_def, applied_overrides) = preprocess.load_scenario_overrides(
+                model_def_dict, scenario, override_dict, **kwargs
             )
+
             self._init_from_model_def_dict(
                 model_def, applied_overrides, scenario, data_source_dfs
             )
+            self.math = preprocess.ModelMath(
```

Comment:
Honestly? No, I do not think so. I like the idea of just moving stuff to `backend.__init__` above:
- Create `ModelMath` with `add_math`.
- Add mode (which is kind of redundant now...)

`Model` will have no math until you build, but maybe that's good?

Reply:
I think it's ok to not have any math until you build, so long as we have a parallell method to build and write the math docs. Then `calliope.Model` could just have `applied_math`, like `applied_overrides` in some way. It tells you what was applied but doesn't give you the sense that you can edit the math dict and it will change the built model in some way.